R/CohortDefinition.R

```diff
@@ -685,20 +671,22 @@ getCohortGenerationOutput <- function(baseUrl, cohortId, sourceKey, mode = 0) {
   if (length(json$inclusionRuleStats) == 0) {
     results$inclusionRuleStats <- NULL
   } else {
-    results$inclusionRuleStas <- json$inclusionRuleStats %>% 
-      jsonlite::toJSON() %>% D
-      jsonlite::fromJSON(simplifyVector = T, simplifyDataFrame = T, flatten = T)
+    inclusionRuleStats <- lapply(json$inclusionRuleStats, tibble::as_tibble)
+    results$inclusionRuleStats <- do.call("rbind", inclusionRuleStats)
   }
   if (length(json$treemapData) == 0) {
     results$treemapData <- NULL
   } else {
-    treeMapResult <- list(name = c(), size = c())
-    treeMapResult <- .flattenTree(json$treemapData, treeMapResult)
-    json$treemapData <- dplyr::tibble(bits = treeMapResult$name, size = treeMapResult$size)
-    json$treemapData$size <- as.integer(json$treemapData$size)
-    json$treemapData$SatisfiedNumber = stringr::str_count(string = json$treemapData$bits, pattern = '1')
-    json$treemapData$SatisfiedRules = stringr::str_locate_all(string = json$treemaDpData$bits, pattern = '1') %>% 
-      paste()
+    results$treemapData <- jsonlite::fromJSON(json$treemapData, simplifyDataFrame = FALSE)
```

Comment:
wouldnt this return a tree of nested lists? 
.flattenTree was written to convert the nested tree to a dataframe

Reply:
Without .flattenTree - a nested list like this is returned - making it hard to use. 

![image](https://user-images.githubusercontent.com/13936600/82147801-44447780-981e-11ea-8afa-9ea1646395c2.png)


.flattenTree helped convert the nested list to a dataframe